fuel-crypto/src/public.rs

```diff
@@ -236,4 +165,13 @@ mod use_std {
                 .and_then(PublicKey::try_from)
         }
     }
+
+    /// Check if the public key byte representation is in the curve.
+    fn public_key_bytes_valid(bytes: &[u8; PublicKey::LEN]) -> Result<(), Error> {
+        let mut public_with_flag = [0u8; UNCOMPRESSED_PUBLIC_KEY_SIZE];
+        public_with_flag[1..].copy_from_slice(bytes);
+        secp256k1::PublicKey::from_slice(&public_with_flag)
+            .map(|_| ())
+            .map_err(|_| Error::InvalidPublicKey)
```

Comment:
TODO: The original `is_slice_in_curve_unchecked` function would also leave the first byte zeroed before invoking the `secp256k1` pubkey parser like this, so I've carried over the behaviour in this function. However, it looks like we should possibly be setting the first byte to `SECP_UNCOMPRESSED_FLAG`? Should investigate or open an issue before landing.